crates/typst-syntax/src/parser.rs

```diff
@@ -1165,8 +1175,11 @@ fn parenthesized_or_array_or_dict(p: &mut Parser) -> SyntaxKind {
 /// State for array/dictionary parsing.
 struct GroupState {
     count: usize,
+    /// A small optimization for if this is _just_ parentheses, `()`.
```

Comment:
:+1:
I think I was confusing this with the if-statement in `expr_with_paren` for parsing assignments. That if-statement checks that an `=` follows and that the kind is not `Parenthesized`. But this causes us to handle some patterns inconsistently, e.g. both `(x) = 1` and `(_) = 2` give an error in 0.12, but we accept `let (x) = 3`. I assume this is to avoid ambiguity with single length arrays, but it's not fully consistent. Should we change this?

Reply:
what change do you propose?

Reply:
I would probably want to disallow `let (x) = {...}` for now and give a hint to add a comma. Even if this is inconsistent with arrow functions, it feels like it would just be subtly incorrect for users trying to deconstruct an array or dict.
Should I make a separate issue to discuss this?

Reply:
Yes, let's keep this separate.